City-of-Helsinki/drupal-module-helfi-tpr
# Uhf 4457 address search

# POC: Search units by address [UHF-4457](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4457)
Adds possibility to add address search to views which lists units. When the user searches units by address, the nearest units are shown first.

## What was done
* Added a new submodule `helfi_address_search` to `helfi_tpr` module.

## How to install
* Make sure your instance is up and running on latest dev branch.
  * It might be easiest to test with the Kasko instance.
* Update the TPR module:
    * `composer require drupal/helfi_tpr:dev-UHF-4457-address-search`
* Go to `/admin/modules` and enable "HELfi Address search".
* Edit some view that lists units, e.g. `vocational_school_search` at Kasko.
* Add `Computed distance` field to the view with default settings.
* Add `Address search` filter to the view with default settings.
* Check which template is used when showing the individual view results. For example, with the Kasko's Vocational School Search, the template file is `hdbt_subtheme/templates/module/helfi_tpr/tpr-unit.html.twig`.
  * Modify the file by adding `{{ content.distance }}` after the h3 title tag.
  * Clear caches: `drush cr`

## How to test
* Go to a page which includes a paragraph (e.g. Vocational School Search) which includes the view block that was just edited.
  * Make sure the paragraph (e.g. Vocational School Search) has some published units listed.
* You should see the address search and when entering a street address, the view results should be ordered so that the nearest results are shown first.
* The distance from the user's street address to the unit's location is shown in meters at the view result rows.
